src/phpDocumentor/Descriptor/Traits/HasSummary.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Descriptor\Traits;
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link https://phpdoc.org
+ */
+
+use phpDocumentor\Descriptor\Interfaces\ElementInterface;
+use phpDocumentor\Descriptor\Interfaces\InheritsFromElement;
+
+use function strtolower;
+use function trim;
+
+trait HasSummary
+{
+    /** @var string $summary A summary describing the function of this element in short. */
+    protected string $summary = '';
```

Comment:
This is a question I also had. I vaguely remember that in the past, we made the decision to make just about everything about Descriptors protected or public so that users could create their own descriptors through extending the existing one. Such a mechanism never really took off and I do not know if someone uses it.. but I am all for tightening it